apps/bluetooth/js/modules/pair_manager.js

```diff
@@ -196,27 +151,31 @@ define(function(require) {
         return;
       }
 
-      this._defaultAdapter.pairingReqs.onpairingconsentreq = 
-        this._onPairingConsentReq.bind(this);
+      this._defaultAdapter.pairingReqs.addEventListener('pairingconsentreq',
+        this._constructPairingInfo.bind(this, 'consent'));
     },
 
     /**
-     * A handler to handle 'onpairingconsentreq' event while it's coming.
+     * A handler to construct pairing info while different pairing requests are
+     * coming.
      *
      * @access private
      * @memberOf PairManager
      */
-    _onPairingConsentReq: function(evt) {
-      Debug('onpairingconsentreq(): Pairing request from ' + 
-            evt.deviceName + ': pairing consent');
-      // TODO: Notify user of just-work pairing if user story is required.
-      throw new Error('Received pairing method "onpairingconsentreq". ' + 
-                      'It would need to implement if user story is required!!');
+    _constructPairingInfo: function(eventName, evt) {
```

Comment:
The name `_constructPairingInfo` sounds like the function simply "construct" pairing info but it actually calls to `_onRequestPairing` and handles the pairing request. It is not easy to understand the purpose of registering `_constructingPairingInfo` when a pairing request event is received.

And if `_onRequestPairing` is not directly registered as a event handler, I would suggest to change its name to `_handlePairingRequest`.

I think we can simply do:
```
this._defaultAdapter.pairingReqs.addEventListener('pairingconsentreq', (event) => {
   this._handlePairingRequest({
     method: 'consent',
     evt: evt
   });
});
```